src/components/animation/animation.ts

```diff
@@ -198,16 +202,12 @@ export default class SlAnimation extends LitElement {
 
   /** Clears all KeyframeEffects caused by this animation and aborts its playback. */
   cancel() {
-    try {
-      this.animation.cancel();
-    } catch {}
```

Comment:
I actually looked this one up specifically. I read the following from [mdn here](https://developer.mozilla.org/en-US/docs/Web/API/Animation/cancel) which made me think that this method can never throw. Maybe I misread though.

> This method doesn't directly throw exceptions; however, if the animation's playState is anything but "idle" when cancelled, the current finished promise is rejected with a DOMException named AbortError.

So this can cause the [finished promise](https://developer.mozilla.org/en-US/docs/Web/API/Animation/finished) to reject, but this method cannot throw itself.

Let me know if I misread and this can actually throw.

Reply:
I tested this out earlier and couldn't get it to throw. I might have made an incorrect assumption from that same line off MDN. Let's leave it.